### PR TITLE
CDC updates

### LINKED ISSUE #3601: changefeedccl: various tests
PR: https://github.com/cockroachdb/cockroach/pull/27962

From release notes:
> `CREATE CHANGEFEED` is now restricted to superusers. [#27962][#27962] {% comment %}doc{% endcomment %}
>
> `CHANGEFEED`s now produce an error when column families are added (instead of returning incorrect results) and when targeting `system` tables (instead of operating with undefined behavior). [#27962][#27962] {% comment %}doc{% endcomment %}
>
> `CREATE CHANGEFEED` now requires an enterprise license when used with Kafka. [#27962][#27962] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by lhirata**
Closes #3601, #3602, #3603, #3605, #3606, #3316, #3752, #3655

### CODE REVIEW

**Comment by rolandcrosby** [comment]
We need to document the time format for WITH CURSOR - it’s an internal cluster timestamp and I don’t think we explicitly talk about the format anywhere. Dan or I can help if you need more details.

File: v2.1/change-data-capture.md [hunk 1]
```diff
@@ -7,29 +7,61 @@ toc: true
...
 
 {{site.data.alerts.callout_danger}}
-**This feature is under active development** and only works for a targeted use case. Please [file a Github issue](file-an-issue.html) if you have feedback on the interface.
-
-In v2.1, CDC will be an enterprise feature and will have a core version.
+**This feature is under active development** and only works for a targeted use case. Please [file a Github issue](file-an-issue.html) if you have feedback on the roadmap.
 {{site.data.alerts.end}}
 
+{{site.data.alerts.callout_info}}
+CDC is an enterprise feature. There will be a core version in v2.2.
```
Comment (rolandcrosby) [inline v2.1/change-data-capture.md]: Do we generally commit to when future versions are coming, or do we want to say "a core version is planned for a future release"?

**Comment by lhirata** [comment]
@danhhz @jseldess - added info about high-water timestamps and made edits based on the last round of feedback. can you review plz?

@Amruta-Ranade - you can take a quick look too (if you want). i think i added enough info to cover #3655.

**Comment by jseldess** [comment]
___
*[v2.1/change-data-capture.md, line 14 at r6](https://reviewable.io/reviews/cockroachdb/docs/3669#-LL1XoYYIT89CORaQN0m-r6-14:-LMrl0Lw3ewvbv9OZ8Tp:b-9dfmm5) ([raw file](https://github.com/cockroachdb/docs/blob/761607ef83b9a3022a8f78bbf5e777bda1c75554/v2.1/change-data-capture.md#L14)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

There's been enough interest in a core version of changefeeds that I think it's worth keeping a mention of our intention to have one. This is the first time that the core alternative to an enterprise feature is not available when the enterprise feature was launched, so I think it's breaking new ground a bit
</blockquote></details>

SGTM. In that case, Lauren, please just add a link to the enterprise licensing page.

**Pull request merged**